kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala

```diff
@@ -82,8 +51,24 @@ case class LambdaFunc(parameter: LambdaVar, body: LambdaTerm) extends LambdaTerm
 
   override val freeVars: Set[String] = body.freeVars - parameter.name
 
+  private[ast] def nestedFunctions(): (List[LambdaVar], LambdaTerm) = {
+    @tailrec
+    def go(func: LambdaFunc, acc: List[LambdaVar]): (List[LambdaVar], LambdaTerm) =
```

Comment:
That's why I made `nestedFunctions` a member of `LambdaFunc`. It is not possible to invoke that `go` with something that is not `LambdaFunc`. Please correct me if I'm wrong.